src/event_handler.rs

```diff
@@ -504,63 +511,148 @@ pub async fn handle_event_logs(
+            sleep(Duration::from_secs(RESEND_TXN_INTERVAL)).await;
+
+            // Get the transaction receipt for the pending transaction to check if it's still pending or been dropped
+            let http_rpc_client = app_state.http_rpc_client.lock().unwrap().clone().unwrap();
+            let Ok(txn_receipt) = http_rpc_client
+                .get_transaction_receipt(pending_txn_data.txn_hash)
+                .await
+            else {
+                continue;
+            };
+
+            // Transaction is confirmed/mined and need not be resent
+            if txn_receipt.is_some() {
+                retry_txn = false;
+                break;
+            }
+
+            // Current txn is dropped/lost and need to be resent with higher gas price
+            pending_txn_data.gas_limit = pending_txn_data.gas_limit + GAS_LIMIT_BUFFER;
+            pending_txn_data.gas_price = U256::from(100 + RESEND_GAS_PRICE_INCREMENT_PERCENT)
+                * pending_txn_data.gas_price
+                / 100;
+
+            // Send the replacement txn if the current one is dropped or lost
+            while Instant::now() < pending_txn_data.retry_deadline {
+                let replacement_txn = pending_txn_data.txn_data.clone();
+                let replacement_txn = replacement_txn
+                    .nonce(pending_txn_data.nonce)
+                    .gas_price(pending_txn_data.gas_price)
+                    .gas(pending_txn_data.gas_limit);
+
+                let pending_txn = replacement_txn.send().await;
+                let Ok(pending_txn) = pending_txn else {
+                    let error_string = format!("{:?}", pending_txn.unwrap_err());
+
+                    // Retry after a delay if connection failed
+                    if error_string.contains("code: -32000")
+                        && (error_string.contains("connection") || error_string.contains("network"))
+                    {
+                        sleep(Duration::from_millis(200)).await;
+                        continue;
+                    }
+
+                    // Retry with a higher gas price if the replacement txn is underpriced
+                    if error_string.contains("code: -32000")
+                        && (error_string.contains("replacement transaction underpriced")
+                            || error_string.contains("transaction underpriced"))
+                    {
+                        pending_txn_data.gas_price =
+                            U256::from(100 + RESEND_GAS_PRICE_INCREMENT_PERCENT)
+                                * pending_txn_data.gas_price
+                                / 100;
+
+                        sleep(Duration::from_millis(10)).await;
+                        continue;
+                    }
+
+                    // Break in case of other errors like "nonce too low" or "execution reverted" or "insufficient funds" etc.
+                    retry_txn = false;
```

Comment:
this can be an issue. Hitting this line mean, txn sender is blocked forever, the current nonce wouldn't be filled any further and queue would just increase. It won't even send dummy txn for current nonce, as L594 would not be true.